kratos/input_output/cad_json_input.h

```diff
@@ -0,0 +1,496 @@
+#if !defined(KRATOS_CAD_JSON_INPUT_INCLUDED )
+#define  KRATOS_CAD_JSON_INPUT_INCLUDED
+
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/io.h"
+#include "includes/kratos_parameters.h"
+#include "includes/model_part.h"
+#include "includes/node.h"
+
+// Geometries
+#include "geometries/nurbs_curve_geometry.h"
+#include "geometries/nurbs_surface_geometry.h"
+#include "geometries/brep_surface.h"
+#include "geometries/brep_curve_on_surface.h"
+#include "geometries/geometry.h"
+
+namespace Kratos
+{
+
+  ///@name Kratos Classes
+  ///@{
+  /// Short class definition.
+  /** Gives IO capabilities for Nurbs based Brep models in the JSON format defined in
+  https://amses-journal.springeropen.com/articles/10.1186/s40323-018-0109-4.
+  */
+    template<class TNodeType = Node<3>, class TEmbeddedNodeType = Point>
+    class KRATOS_API(KRATOS_CORE) CadJsonInput //: public IO
+    {
+    public:
+
+        ///@}
+        ///@name Type Definitions
+        ///@{
+
+        /// Pointer definition of CadJsonInput
+        KRATOS_CLASS_POINTER_DEFINITION(CadJsonInput);
+
+        typedef std::size_t SizeType;
+        typedef std::size_t IndexType;
+
+        typedef PointerVector<TNodeType> ContainerPointType;
+        typedef PointerVector<TEmbeddedNodeType> ContainerPointEmbeddedType;
+
+        typedef NurbsSurfaceGeometry<3, ContainerPointType> NurbsSurfaceType;
+        typedef BrepSurface<ContainerPointType, ContainerPointEmbeddedType> BrepSurfaceType;
+        typedef BrepCurveOnSurface<ContainerPointType, ContainerPointEmbeddedType> BrepCurveOnSurfaceType;
+
+        typedef DenseVector<typename BrepCurveOnSurfaceType::Pointer> BrepCurveOnSurfaceLoopType;
+        typedef DenseVector<DenseVector<typename BrepCurveOnSurfaceType::Pointer>> BrepCurveOnSurfaceLoopArrayType;
+
+        ///@}
+        ///@name Life Cycle
+        ///@{
+
+        /// Constructor.
+        CadJsonInput(
+            const std::string & rDataFileName)
+        {
+            std::ifstream cad_json_file;
+            cad_json_file.open(rDataFileName);
+
+            std::string cad_json_string((std::istreambuf_iterator<char>(cad_json_file)),
+                std::istreambuf_iterator<char>());
+
+            mCadJsonParameters = Parameters(cad_json_string);

```

Comment:
Instead of reading the file in C++, I would instead read it in python and then pass the `Parameters` to the IO

Reply:
actually maybe even better if you pass both the parameters and the modelpart to `ReadGeometryModelPart`, to me it is not clear that you need to save it
Unless you will add sth for which you will need it

Reply:
I had it like that before. @pooyan-dadvand preferred it like this. I personally don't have a deep opinion about it. Would be also possible to have both/ all options

Reply:
@tteschemacher you mean the saving of the parameters as member?
I guess this is to be consistent with the `ModelpartIO`
@pooyan-dadvand I don't know the reason, but to me in this case I don't see the use
Could you please explain?